### PR TITLE
bpo-44977: Deprecate delegation of int to __trunc__

### PR DESCRIPTION
**Pull request by ZackerySpytz**
Calling `int(a)` when `type(a)` implements `__trunc__` but not `__int__` or `__index__` now raises a DeprecationWarning.

https://bugs.python.org/issue44977

### CODE REVIEW

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

File: Doc/whatsnew/3.11.rst [hunk 1]
```diff
@@ -458,6 +458,11 @@ Deprecated
   as deprecated, its docstring is now corrected).
   (Contributed by Hugo van Kemenade in :issue:`45837`.)
 
+* The delegation of :func:`int` to :meth:`__trunc__` is now deprecated. Calling
```
Comment (kumaraditya303) [inline Doc/whatsnew/3.11.rst:461]:
```suggestion
* The delegation of :class:`int` to :meth:`__trunc__` is now deprecated. Calling
```

File: Misc/NEWS.d/next/Core and Builtins/2022-01-30-18-23-08.bpo-44977.BQV_zS.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+The delegation of :func:`int` to :meth:`__trunc__` is now deprecated.
```
Comment (kumaraditya303) [inline Misc/NEWS.d/next/Core and Builtins/2022-01-30-18-23-08.bpo-44977.BQV_zS.rst:1]:
```suggestion
The delegation of :class:`int` to :meth:`__trunc__` is now deprecated.
```

**Review by kumaraditya303 — CHANGES_REQUESTED** [review]

File: Doc/whatsnew/3.11.rst (hunk 1 above)
Comment (serhiy-storchaka) [reply Doc/whatsnew/3.11.rst:461]: `int()` is referred as a function here, not a class.

The current text is correct.

**Pull request merged**